UCSCore/src/main/java/it/cnr/iit/ucs/core/UCSBuilder.java

```diff
@@ -53,42 +42,39 @@
  * This class is in charge of instantiating all the components.
  *
  * @author Antonio La Marra, Alessandro Rosetti
- *
  */
-@Component
-public class UCSRest implements UCSInterface {
-
-    private static final Logger log = Logger.getLogger( UCSRest.class.getName() );
-
-    private AbstractContextHandler contextHandler;
-    private AbstractRequestManager requestManager;
-    private ObligationManagerInterface obligationManager;
-    private SessionManagerInterface sessionManager;
-    private PDPInterface pdp;
-    private PAPInterface pap;
-    private List<PIPBase> pipList = new ArrayList<>();
-    private HashMap<String, PEPInterface> pepMap = new HashMap<>();
+public class UCSBuilder {
 
-    private boolean initialised = false;
+    private static final Logger log = Logger.getLogger( UCSBuilder.class.getName() );
 
-    @Autowired
+    private UCSCore ucsCore;
     private UCSProperties properties;
 
-    @PostConstruct
-    private void init() {
+    public UCSBuilder() {
+        ucsCore = new UCSCore();
```

Comment:
why do this, when it is auto wired?

Reply:
It's not autowired since we don't use spring in UCSCore

Reply:
OK I see. I miss-read it

Reply:
OK I see now

Reply:
Now it's called UCSCoreBuilder but then it will be called UCSCoreServiceBuilder